pkg/controller/releasemanager/plan/syncApp.go

```diff
@@ -126,7 +123,14 @@ func (p *SyncApp) releaseMatches(log logr.Logger, port picchuv1alpha1.PortInfo)
 			return matches
 		}
 		gateway = []string{p.PrivateGateway}
-		hosts = append(hosts, p.defaultHost())
+		hosts = append(hosts, fmt.Sprintf("%s.%s", p.Namespace, p.DefaultDomain)) // default host
+
+		portName := port.Name
+		if portName == "" {
+			portName = strconv.Itoa(int(port.Port))
+		}
+		hosts = append(hosts, fmt.Sprintf("%s-%s.%s", p.Namespace, portName, p.DefaultDomain))
```

Comment:
```suggestion
		if port.Name != "" {
			hosts = append(hosts, fmt.Sprintf("%s-%s.%s", p.Namespace, port.Name, p.DefaultDomain))
		}
```
how about only supporting this for named ports?